### PR TITLE
Feature: update user profile

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

Allows a user to change (only their) password, email, name. Also allows retrieving the token from the frontend, a pet peeve of mine.

Screenshot 2023-11-26 at 7 41 17 AM

https://github.com/paperless-ngx/paperless-ngx/assets/4887959/e141407f-f941-443b-a19b-43c496df20a1

Fixes #3230

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain):

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Review by stumpylog — CHANGES_REQUESTED** [review]
Still need to actually test, but had a though about the backend anyway

File: src/documents/tests/test_api.py [hunk 1]
```diff
@@ -5793,3 +5794,84 @@ def test_api_create_consumption_template_with_mailrule(self):
         self.assertEqual(ConsumptionTemplate.objects.count(), 2)
         ct = ConsumptionTemplate.objects.get(name="Template 2")
         self.assertEqual(ct.sources, [int(DocumentSource.MailFetch).__str__()])
+
+
+class TestApiProfile(DirectoriesMixin, APITestCase):
```
[the remark this answers is not in the document]
Comment (shamoon) [reply src/documents/tests/test_api.py:5799]: Yea it is, hard to find stuff sometimes. Maybe `test_api_profile`?

[the remark this answers is not in the document]
Comment (stumpylog) [reply src/documents/tests/test_api.py:5799]: Yeah, I was going to get around to it sometime, but I planned to keep things as `test_api_*.py` so they stay grouped

**Comment by shamoon** [comment]
Let me know if good with the changes, I honestly think its not possible that the request doesnt have 'user' so I left it off, but let me know if Im wrong about that (and if so, how to write a test that would repro it). I did add test for logged out to confirm the 401

**Comment by stumpylog** [comment]
One thing I'm seeing, after the logout from changing the password, the page after i log back in is consistently a 404.  I can't tell from the network tools what it's trying to load though, nothing there is returning a 404

**Comment by shamoon** [comment]
Ah, this reminds me I had to include a manual logout for the users component when the pw changes, I bet that'll do it. Good catch, thanks.

**Comment by shamoon** [comment]
Annoying thing about my testing setup is it isnt really good for testing that kinda stuff, I'll double-check but this should now log you out just like under 'users & groups'

**Comment by shamoon** [comment]
Actually it turns out that issue applies to the other (current) way to change a password too. And recreating it is very easy, if you go to `/accounts/logout/?next=/accounts/login/` (which is what this currently does) it'll happen. The solution seems simple enough which is to change it to `/accounts/logout/?next=/accounts/login/?next=/` which, though it looks wrong, seems ok. Basically its https://stackoverflow.com/questions/13186693/why-second-user-login-redirects-me-to-accounts-profile-url-in-django

Also fixed another small thing I noticed, hopefully this should be good now.

**Review by stumpylog — APPROVED** [review]
Yep, redirecting works great now

**Pull request merged**